python-backend/app/api/parties/party/resources/party_resource.py

```diff
@@ -54,113 +51,59 @@ class PartyResource(Resource, UserMixin, ErrorMixin):
-        if party_guid:
-            try:
-                party = Party.find_by_party_guid(party_guid)
-            except DBAPIError:
-                return self.create_error_payload(422, 'Invalid Party guid'), 422
-            if party:
-                return party.json()
-            else:
-                return self.create_error_payload(404, 'Party not found'), 404
-        else:
-            search_term = request.args.get('search')
-            search_type = request.args.get('type').upper() if request.args.get('type') else None
-            items_per_page = request.args.get('per_page', 25, type=int)
-            page = request.args.get('page', 1, type=int)
-            parties = Party.query
-            if search_term:
-                if search_type in ['PER', 'ORG']:
-                    parties = Party.search_by_name(search_term, search_type,
-                                                   self.PARTY_LIST_RESULT_LIMIT)
-                else:
-                    parties = Party.search_by_name(
-                        search_term, query_limit=self.PARTY_LIST_RESULT_LIMIT)
-
-            paginated_parties, pagination_details = apply_pagination(parties, page, items_per_page)
-            if not paginated_parties:
-                self.raise_error(404, 'No parties found'), 404
-            parties = paginated_parties.all()
-            relationships = request.args.get('relationships')
-            relationships = relationships.split(',') if relationships else []
-            return {
-                'parties': list(map(lambda x: x.json(relationships=relationships), parties)),
-                'current_page': pagination_details.page_number,
-                'total_pages': pagination_details.num_pages,
-                'items_per_page': pagination_details.page_size,
-                'total': pagination_details.total_results,
-            }
-
-    @api.expect(parser)
-    @requires_role_mine_create
-    def post(self, party_guid=None):
-        if party_guid:
-            raise BadRequest('Unexpected party id in Url.')
-        data = PartyResource.parser.parse_args()
-
-        party = Party.create(
-            data.get('party_name'),
-            data.get('phone_no'),
-            data.get('party_type_code'),
-            # Nullable fields
-            email=data.get('email'),
-            first_name=data.get('first_name'),
-            phone_ext=data.get('phone_ext'),
-            suite_no=data.get('suite_no'),
-            address_line_1=data.get('address_line_1'),
-            address_line_2=data.get('address_line_2'),
-            city=data.get('city'),
-            sub_division_code=data.get('sub_division_code'),
-            post_code=data.get('post_code'))
+    @api.marshal_with(PARTY, code=200)
+    def get(self, party_guid):
+        try:
+            uuid.UUID(party_guid)
+        except ValueError:
+            raise BadRequest('Invalid Party guid')
 
+        party = Party.find_by_party_guid(party_guid)
```

Comment:
I think we should handle the formatting of the guid in the find helper. find_by_party_guid should return None or throw an error if it is malformed... thoughts?

Reply:
Oh, yeah. Definitely. I did that for Variances. Leaving it here was an accident. Thanks!